### PR TITLE
[conformance] Enable local cache using in SubgraphsDumper tool

### PR DESCRIPTION
**Pull request by sbalandi**
### Details:
- *Enable correct work with local cache in SubraphsSumper*

subgraphs_dumper_arch_diaram

### Tickets:
 - *CVS-117950*

### CODE REVIEW

File: src/tests/functional/plugin/conformance/subgraphs_dumper/include/cache/cache.hpp [hunk 1]
```diff
@@ -18,7 +18,7 @@ namespace subgraph_dumper {
 class ICache {
 public:
     virtual void update_cache(const std::shared_ptr<ov::Model>& model,
-                              const std::string& source_model, bool extract_body = true) {};
+                              const std::string& source_model, bool extract_body = true, bool from_cache = false) {};
```
Comment (sbalandi) [inline src/tests/functional/plugin/conformance/subgraphs_dumper/include/cache/cache.hpp:21]: on string std::string m_serialization_dir = ".";  - add subdirectory operation / subgraph (m_serialization_dir/m_cache_dir)

File: src/tests/functional/plugin/conformance/subgraphs_dumper/src/main.cpp [hunk 2]
```diff
@@ -53,7 +53,7 @@ int main(int argc, char *argv[]) {
     if (!FLAGS_local_cache.empty()) {
```
Comment (sbalandi) [inline src/tests/functional/plugin/conformance/subgraphs_dumper/src/main.cpp:53]: for (auto& cache : caches) {

File: src/tests/functional/plugin/conformance/subgraphs_dumper/include/cache/meta/meta_info.hpp [hunk 3]
```diff
@@ -28,7 +28,7 @@ class MetaInfo {
     std::map<std::string, ModelInfo> get_model_info() const;
     std::string get_any_extractor() const { return *extractors.begin(); }
 
-    static MetaInfo read_meta_from_file(const std::string& meta_path);
+    static MetaInfo read_meta_from_file(const std::string& meta_path, size_t model_priority = 1);
```
Comment (iefode) [inline src/tests/functional/plugin/conformance/subgraphs_dumper/include/cache/meta/meta_info.hpp:31]: The best option is to add model_priority to meta directly

File: src/tests/test_utils/functional_test_utils/layer_tests_summary/rename_conformance_ir.py [hunk 4]
```diff
@@ -93,48 +95,38 @@ def is_report_op(op_name:str):
...
             str(input_node.get_type_info().name) + str(input_node.get_type_info().version_id)
+
     for output in node.outputs():
         output_node = output.get_node()
-        len_shape = None
+
+        shape_str = ""
         try:
-            len_shape = len(output.get_partial_shape())
+            partial_shape = output.get_partial_shape()
```
Comment (iefode) [inline src/tests/test_utils/functional_test_utils/layer_tests_summary/rename_conformance_ir.py:120]: revert the changes? except of `.remove file`

File: src/tests/functional/plugin/conformance/subgraphs_dumper/src/main.cpp [hunk 5]
```diff
@@ -48,34 +48,37 @@ int main(int argc, char *argv[]) {
...
-            auto& key = model_status.first;
-            auto& value = model_status.second;
-            if (not_read_model.first == key) {
-                value.insert(value.end(), not_read_model.second.begin(), not_read_model.second.end());
-            }
-            if (this_cache_model_status.count(key)) {
-                value.insert(value.end(), this_cache_model_status[key].begin(), this_cache_model_status[key].end());
+        // Upload previously cached graphs to cache
+        if (!FLAGS_local_cache.empty()) {
+            auto cached_ops = find_models(local_cache_dirs, cache->get_cache_subdir_regexp());
```
Comment (iefode) [inline src/tests/functional/plugin/conformance/subgraphs_dumper/src/main.cpp:53]: ov::util::join_path([local_cache_dirs, cache->get_cache_subdir_regexp()])

Comment (sbalandi) [reply src/tests/functional/plugin/conformance/subgraphs_dumper/src/main.cpp]: done

File: src/tests/functional/plugin/conformance/subgraphs_dumper/include/cache/op_cache.hpp [hunk 6]
```diff
@@ -36,10 +36,15 @@ class OpCache : public ICache {
...
 
+    virtual std::string get_cache_subdir_regexp() override {
+        return ".*" + m_cache_subdir + ".*";
+    };
+
 protected:
     std::map<std::shared_ptr<ov::Node>, MetaInfo> m_ops_cache;
     static std::shared_ptr<OpCache> m_cache_instance;
     MatchersManager m_manager = MatchersManager();
+    std::string m_cache_subdir = "operation";
```
[the remark this answers is not in the document]
Comment (sbalandi) [reply src/tests/functional/plugin/conformance/subgraphs_dumper/include/cache/op_cache.hpp]: done

File: src/tests/functional/plugin/conformance/subgraphs_dumper/include/cache/cache.hpp [hunk 7]
```diff
@@ -18,9 +18,10 @@ namespace subgraph_dumper {
...
     virtual void serialize_cache() {};
     virtual void reset_cache() {};
+    virtual std::string get_cache_subdir_regexp() {};
```
[the remark this answers is not in the document]
Comment (sbalandi) [reply src/tests/functional/plugin/conformance/subgraphs_dumper/include/cache/cache.hpp]: removed

File: src/tests/functional/plugin/conformance/subgraphs_dumper/include/cache/meta/meta_info.hpp (hunk 3 above)
Comment (sbalandi) [reply src/tests/functional/plugin/conformance/subgraphs_dumper/include/cache/meta/meta_info.hpp:31]: what do you mean ? I need to add method to update model_priority here ?

**Review by iefode — APPROVED** [review]

**Comment by iefode** [comment]
@sbalandi Make CI green and I will merge it (check that SubgraphsDumperTests are OK) :)
